primefaces/src/main/resources/META-INF/resources/primefaces/datepicker/0-datepicker.js

```diff
@@ -1857,16 +1867,205 @@
             this.panel.off('click.datePicker-clearButton', clearButtonSelector).on('click.datePicker-clearButton', clearButtonSelector, null, this.onClearButtonClick.bind($this));
 
             var dateSelector = '.ui-datepicker-calendar td a';
-            this.panel.off('click.datePicker-date', dateSelector).on('click.datePicker-date', dateSelector, null, function (event) {
+            this.panel.off('click.datePicker-date keydown..datePicker-date', dateSelector).on('click.datePicker-date', dateSelector, null, function (event) {
                 if ($this.monthsMetadata) {
                     var dayEl = $(this),
                         calendarIndex = dayEl.closest('.ui-datepicker-group').index(),
                         weekIndex = dayEl.closest('tr').index(),
                         dayIndex = dayEl.closest('td').index() - ($this.options.showWeek ? 1 : 0);
                     $this.onDateSelect(event, $this.monthsMetadata[calendarIndex].dates[weekIndex][dayIndex]);
                 }
+            }).on('keydown.datePicker-date', dateSelector, null, function (event) {
+                var dayEl = $(this),
+                        calendarIndex = dayEl.closest('.ui-datepicker-group').index(),
+                        weekIndex = dayEl.closest('tr').index(),
+                        dayIndex = dayEl.closest('td').index() - ($this.options.showWeek ? 1 : 0);
+                 $this.onDateKeyDown(event, $this.monthsMetadata[calendarIndex].dates[weekIndex][dayIndex]);
             });
         },
+        
+        onDateKeyDown: function(event, dateMeta) {
+            if (this.options.disabled || !dateMeta.selectable) {
+                if (event) {
+                    event.preventDefault();
+                }
+                return;
+            }
+
+            var $this = this;
+            var currentElement = $(event.currentTarget);
+            var $tabbableElements = $this.panel.find('a:tabbable');
+            var currentIndex = $tabbableElements.index(currentElement);
+            switch (event.key) {
+                case 'Enter':
```

Comment:
Don't we have a constant with key names?

Reply:
I think you might be the thinking of the old Jquery constants for key codes but since these keys are the actual key names it makes the code readable without constants. For example it was keycode === 27 which would be $ui.keycode.ESCAPE.   

but now none of that is necessary as event.key === 'Escape' reads much better and is supported in all browsers.

Reply:
It's just that, in Java, I never repeat the same string over and over. And I don't like things to be stringly typed. 😁